openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/EncounterResource.java

```diff
@@ -39,43 +40,34 @@
  * point in time (referred to as the "snapshot time").
  * @see AbstractReadOnlyResource
  */
-// TODO: Merge with PatientResource and let clients use a query parameter to
-// indicate whether encounter data should be included with each returned patient.
 @Resource(name = RestController.REST_VERSION_1_AND_NAMESPACE + "/encounters",
     supportedClass = Patient.class, supportedOpenmrsVersions = "1.10.*,1.11.*")
 public class EncounterResource
-    extends AbstractReadOnlyResource<Patient> implements Creatable {
+    extends AbstractReadOnlyResource<Encounter> implements Creatable {
     private final PatientService patientService;
-    private final EncounterService encounterService;
+    private final ProjectBuendiaService buendiaService;
 
     public EncounterResource() {
-        super("patient", Representation.DEFAULT);
+        super("encounter", Representation.DEFAULT);
         patientService = Context.getPatientService();
-        encounterService = Context.getEncounterService();
+        buendiaService = Context.getService(ProjectBuendiaService.class);
     }
 
     /**
-     * Returns all patients.  The retrieved records will be filled in with
-     * each patient's encounter and observation data by
-     * {@link #populateJsonProperties(Patient, RequestContext, SimpleObject, long)}
-     * on its way to becoming JSON that is sent to the client.
-     * @param context      unused here; see populateJsonProperties() for details
+     * Returns all encounters observed after the "since" parameter, or since the beginning of time
+     * if the "since" parameter is not set.
+     * @param context      used to obtain the "since" parameter for incremental sync.
      * @param snapshotTime unused here; see populateJsonProperties() for details
      * @see AbstractReadOnlyResource#search(RequestContext)
      */
-    @Override public List<Patient> searchImpl(RequestContext context, long snapshotTime) {
-        String patientUuid = context.getParameter("patient_uuid");
-        if (patientUuid != null) {
-            Patient patient = patientService.getPatientByUuid(patientUuid);
-            if (patient == null) {
-                throw new ObjectNotFoundException();
-            }
-            List<Patient> patients = new ArrayList<>();
-            patients.add(patient);
-            return patients;
-        } else {
-            return patientService.getAllPatients();
+    @Override public List<Encounter> searchImpl(RequestContext context, long snapshotTime) {
+        Date syncFrom;
+        try {
+            syncFrom = RequestUtil.getSyncFromDate(context);
```

Comment:
Maybe. My intended design is to:

- Filter by the `since=` parameter, which is a timestamp previously supplied by the server.
- Paginate using a `page=` parameter, which is independent from modification time
- Sort the data by UUID, to ensure stable ordering.

So the resultant SQL query approximately looks like:

    WHERE date_modified >= [since parameter] ORDER BY uuid LIMIT [page_num*page_size],[(page_num + 1)*page_size]

So it doesn't matter that UUIDs aren't monotonically increasing, just that they're unique, which provides stable ordering such that the LIMIT clause provides consistent results across consecutive queries.

Does that make more sense?